nixos/modules/services/continuous-integration/woodpecker/agent.nix

```diff
@@ -0,0 +1,78 @@
+{
+  config
+  , lib
+  , pkgs
+  , ...
+}:
+
+let
+  cfg = config.services.woodpecker-agent;
+in {
+  meta.maintainers = [ lib.maintainers.janik ];
+
+  options = {
+    services.woodpecker-agent = {
+      enable = lib.mkEnableOption (lib.mdDoc "Woodpecker-Agent executes tasks from the Woodpecker-Server");
+      package = lib.mkPackageOptionMD pkgs "woodpecker-agent" { };
+      stateDir = lib.mkOption {
+        default = "/var/lib/woodpecker";
+        type = lib.types.str;
+        description = lib.mdDoc "woodpecker data directory.";
+      };
+
+      environment = lib.mkOption {
+        default = {};
+        type = lib.types.attrsOf lib.types.str;
+        example = lib.literalExpression
+          ''
+            {
+              WOODPECKER_AGENT_SECRET="your-shared-secret-goes-here";
+              WOODPECKER_SERVER="localhost:9000";
+              WOODPECKER_BACKEND="docker";
+              DOCKER_HOST = "unix:///run/user/0/podman/podman.sock";
+            }
+          '';
```

Comment:
You right, I'm going to implement that. but would still like to have the example variable there, since you could also use it with sops nix or some other secret management and then that should be fine.

Reply:
Woodpecker supports secret reading from a file, so it may be sufficient to just update the description to use `WOODPECKER_AGENT_SECRET_FILE` instead.

Reply:
Maybe I'm misunderstanding something but sops for example can save values in a file that are encrypted and get decrypeted with an age or ssh key when build the config. there is also other secret management tools, but they would still be secret then even when stored together with other config?

Reply:
Okay, I will look at implementing it the way you are suggesting.